src/call/mod.rs

```diff
@@ -646,29 +648,20 @@ impl SinkBase {
         &mut self,
         call: &mut C,
         t: &T,
-        mut flags: WriteFlags,
+        flags: WriteFlags,
         ser: SerializeFn<T>,
     ) -> Result<()> {
-        // `start_send` is supposed to be called after `poll_ready` returns ready.
-        assert!(self.batch_f.is_none());
+        // If there is already a buffered message waiting to be sent, set `buffer_hint` to true to indicate
```

Comment:
Need to check `self.send_metadata` first.

Reply:
will be checked in the actual function.

https://github.com/tikv/grpc-rs/pull/469/files#diff-8c8308f2c854f39735f331e70f94d8d2R705-R708

Reply:
Yes, but I prefer to keep it into `start_send`, because I think it's more clear.